src/Tools/IdeCoreBenchmarks/FindReferencesBenchmarks.cs

```diff
@@ -23,77 +27,86 @@ namespace IdeCoreBenchmarks
-            // anyways  to perform the search on all the equivalent symbols from them.  So the end perf cost is the
-            // same.
-            var project = solution.Projects.First(p => p.AssemblyName == "Microsoft.CodeAnalysis.CSharp");
-
-            var start = DateTime.Now;
-            var compilation = await project.GetCompilationAsync();
-            Console.WriteLine("Time to get first compilation: " + (DateTime.Now - start));
-            var type = compilation.GetTypeByMetadataName("Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser");
-            if (type == null)
-                throw new Exception("Couldn't find type");
-
-            start = DateTime.Now;
-            var references = await SymbolFinder.FindReferencesAsync(type, solution);
-            Console.WriteLine("Time to find-refs: " + (DateTime.Now - start));
-            var refList = references.ToList();
-            Console.WriteLine($"References count: {refList.Count}");
-            var locations = refList.SelectMany(r => r.Locations).ToList();
-            Console.WriteLine($"Locations count: {locations.Count}");
+            try
+            {
+                // QueryVisualStudioInstances returns Visual Studio installations on .NET Framework, and .NET Core SDK
+                // installations on .NET Core. We use the one with the most recent version.
+                var msBuildInstance = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(x => x.Version).First();
+
+                MSBuildLocator.RegisterInstance(msBuildInstance);
+
+                var roslynRoot = Environment.GetEnvironmentVariable(Program.RoslynRootPathEnvVariableName);
+                var solutionPath = Path.Combine(roslynRoot, @"C:\github\roslyn\Compilers.sln");
+
+                if (!File.Exists(solutionPath))
+                    throw new ArgumentException("Couldn't find Compilers.sln");
+
+                Console.Write("Found Compilers.sln: " + Process.GetCurrentProcess().Id);
+
+                var assemblies = MSBuildMefHostServices.DefaultAssemblies
+                    .Add(typeof(AnalyzerRunnerHelper).Assembly)
+                    .Add(typeof(FindReferencesBenchmarks).Assembly);
+                var services = MefHostServices.Create(assemblies);
+
+                var workspace = MSBuildWorkspace.Create(new Dictionary<string, string>
+                {
+                    // Use the latest language version to force the full set of available analyzers to run on the project.
+                    { "LangVersion", "9.0" },
+                }, services);
+
+                if (workspace == null)
+                    throw new ArgumentException("Couldn't create workspace");
+
+                workspace.TryApplyChanges(workspace.CurrentSolution.WithOptions(workspace.Options
+                    .WithChangedOption(StorageOptions.Database, StorageDatabase.SQLite)
+                    .WithChangedOption(StorageOptions.DatabaseMustSucceed, true)));
```

Comment:
Is there a reason some of this isn't refactored into helpers?

Reply:
because it's benchmark code.  the purpose of it is to allow me to rapidly change and rerun it and get results.  the more layers of abstraction and sharing, teh more difficult it is to do that.